### DISCUSSION TITLE
Difference with 1.0.5

### DISCUSSION THREAD

**Discussion by geckotdf** [ian-travers/nfsu-server#7] [Q&A]
Hi!
What the diference between this version and the 1.0.5 of Harpy?

Thanks!

**Comment by ian-travers** [comment]
The main goal was to implement REP points system and statistics like in the original game.

**Comment by geckotdf** [comment]
I see,
So I can replace the 1.0.5 with this version with a new rusers.dat and will have the REP Points system working?

**Comment by ian-travers** [comment]
You can try. Then you can check whether it worked or not.

**Comment by casasfernando** [comment]
@geckotdf which version did you end up using? 1.0.5 or 2.6?

@ian-travers looking https://github.com/HarpyWar/nfsuserver/commits/master/ it seems that you merged some commits to that repo but then your last commit is from June 23, 2022 while in this repo the last commit is from August 22, 2022 so I was wondering:

1. Are there any commits from this repo that were not PR'ed to Harpy's repo? If yes, were there any reasons not to create a PR for them?
2. I'm thinking about forking Harpy's repo, merge any commits missing and needed (e.g. any reputation feature fixes or enhancements) from this repo and then create a PR for Harpy's. Would that make sense? I like the idea of keeping all the great work done so far in the same repo in case anyone wants to continue developing nfsuserver.

Looking forward your thoughts!

Thanks

**Comment by geckotdf** [comment]
Hi @casasfernando 
I stick with the 1.0.5 - But only for a few months, then the thing with the server (it's not a server problem) it's the player that host the party, must have the ports open to allow anothers players to connect, and by today ISPs it's quite hard, most people are behind a NAT internet provider, so you don't have a real public IP.
In my particular project I use VPN for this kind of games, similar to Hamachi / GameRanger.

**Reply by casasfernando** [reply]
Thanks @geckotdf for the insights. I’m planning to run the server mainly on a LAN so I think I will not face that sort of problems. At the moment I’m testing 1.0.5 and had no issues but will keep the VPN idea in mind for remote players.

**Reply by geckotdf** [reply]
Perfect, well in the case of LAN the server it's not necessary because someone create the lobby and the rest connect and create the race.
The main idea of the server it's be able to register and have some stats, and REP points, etc.
If you are going to try to build something with a "LAN" enviroment for public users can use projects like https://localtonet.com/
Or just build in any cheap VPS a softether VPN and will make the trick too.

**Reply by casasfernando** [reply]
Isn’t the server needed even for local multiplayer? There is no option in NFSU for multiplayer but the “Play Online” option and selecting it will try to connect to EA servers which are offline. So unless I’m missing something nfsuserver is still required to access the multiplayer options.

If I recall correctly NFSU2 was different and there were both, local multiplayer and online multiplayer options.

**Reply by geckotdf** [reply]
Sorry my mistake, it's true in NFSU 1 need this to lan too because like you say only have Online Internet mode only, I mistake with NFSU2 that have LAN support.
Yes in LAN in NFSU 1 you still need this server to create a "server" to allow players to be seen.
The issue with this method the computer that "host" the server can't run the game.

For example if you create a LAN party one computer must be alone with the server or some computer must have a Virtual Machine in Bridge mode (not NAT mode) with the server inside and everybody will be able to connect.
Because this server use the same ports of the game, so if you run the server in your computer, your game will not be able to connect.
